ethereumj-core/src/main/java/org/ethereum/net/server/ChannelManager.java

```diff
@@ -344,7 +344,7 @@ public void add(Channel peer) {
         newPeers.add(peer);
     }
 
-    public void notifyDisconnect(Channel channel) {
+    public synchronized void notifyDisconnect(Channel channel) {
         logger.debug("Peer {}: notifies about disconnect", channel);
         channel.onDisconnect();
```

Comment:
The same here. There is no need to call those notify methods under the lock. They can potentially have deadlock side effects